### PR TITLE
Event model

### PR DESCRIPTION
**Pull request by sbufan**

### CODE REVIEW

File: events/event/models.py [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-
 class Event(models.Model):
     title = models.CharField(max_length=200)
     start_date = models.DateTimeField(blank=True, null=True)
     end_date = models.DateTimeField(blank=True, null=True)
     location = models.CharField(max_length=200, null=True)
     description = models.TextField(blank=True, null=True)
+    address = models.CharField(max_length=500, null=True)
+    created_date = models.DateTimeField(blank=True, null=True)
+    owner_id = models.BigIntegerField(null=True)
```
Comment (Scandie) [inline events/event/models.py]: @sbufan, can we use ForeignKey for that purpose?

File: events/event/views.py [hunk 2]
```diff
@@ -34,6 +34,31 @@ def get(self, request, pk=None):
...
+            }, status=400)
+        validation_form = EventCreateForm(event_data)
+        if validation_form.is_valid():
+            event = Event()
+            event.title = event_data.get('title')
+            event.start_date = event_data.get('start_date')
+            event.end_date = event_data.get('end_date')
+            event.location = event_data.get('location')
+            event.description = event_data.get('description')
+            event.owner_id = event_data.get('owner_id')
```
Comment (Scandie) [inline events/event/views.py]: @sbufan. No need to expect this data from front-end. In this case you could use:
`event.owner_id = request.user.id`

File: events/event/views.py [hunk 3]
```diff
@@ -34,6 +34,31 @@ def get(self, request, pk=None):
         else:
             return HttpResponseForbidden('Permission denied')
 
+    def post(self, request):
+        try:
+            event_data = json.loads(request.body.decode())
+        except Exception as e:
+            return JsonResponse({
+                "json error": e.message,
```
Comment (Scandie) [inline events/event/views.py]: @sbufan, Not the best way of handling exception, but even in this case, you should use something like this:
`"json error": str(e),`

File: events/event/models.py [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
 
-
 class Event(models.Model):
     title = models.CharField(max_length=200)
     start_date = models.DateTimeField(blank=True, null=True)
     end_date = models.DateTimeField(blank=True, null=True)
     location = models.CharField(max_length=200, null=True)
     description = models.TextField(blank=True, null=True)
+    address = models.CharField(max_length=500, null=True)
+    created_date = models.DateTimeField(blank=True, null=True)
```
Comment (ghost) [inline events/event/models.py]: Add attribute auto_now_add=True. It makes to possible automatically create the date of the event.

File: events/event/forms.py [hunk 5]
```diff
@@ -1,7 +1,38 @@
...
-        exclude = []
\ No newline at end of file
+        exclude = ['participants', 'start_date', 'end_date']
+    start_date = DateValidator()
+    end_date = DateValidator()
+
+    def clean_location(self):
+        data = self.cleaned_data.get('location')
+        try:
+            type(data[0]).__name__ == 'float' and type(data[1]).__name__ == 'float'
```
Comment (yevgenbehen) [inline events/event/forms.py]: I prefer something like this `isinstance(y, float)`

File: events/event/models.py [hunk 6]
```diff
@@ -2,15 +2,24 @@
 from registration.models import User
 from django.db.models.fields.related import ManyToManyField
 from django.contrib.postgres.fields import ArrayField
+from django.utils.timezone import utc
+
+from datetime import datetime
+
+
+
+TZ = utc
```
Comment (yevgenbehen) [inline events/event/models.py]: Why do you need this?

Comment (ghost) [reply events/event/models.py]: Datetimefield in django need datetime python object with timezone (something like this - datetime.datetime(1970, 1, 1, 3, 2, 3, **tzinfo=< UTC >**)). I use it for adding timezone to datetime

File: events/event/forms.py (hunk 5 above)
Comment (ghost) [reply events/event/forms.py]: Changed

File: events/event/models.py (hunk 6 above)
Comment (yevgenbehen) [reply events/event/models.py]: But you can use directly `utc` without redefining it.  I mean `from django.utils.timezone import utc as TZ`. Will it work?

**Review by yevgenbehen — APPROVED** [review]
+1

File: events/event/forms.py [hunk 7]
```diff
@@ -1,7 +1,38 @@
...
+            isinstance(data[0], float) and isinstance(data[1], float)
+        except:
+            raise forms.ValidationError("Enter valid data")
+        return data
+    
+    def clean_end_date(self):
+        start_date = self.cleaned_data.get('start_date')
+        end_date = self.cleaned_data.get('end_date')
+        if float(end_date) - float(start_date) < 60 * 60:
+            raise forms.ValidationError("End date cannot be earlier than Start date")
```
Comment (BorysYudin) [inline events/event/forms.py]: Please add new line

**Review by BorysYudin — APPROVED** [review]

File: events/event/forms.py (hunk 7 above)
Comment (ghost) [reply events/event/forms.py]: Fix

**Pull request merged**